Beginners/Lesson Five/README.md

```diff
@@ -0,0 +1,218 @@
+
+`exit(0)` is a way of forcing your application to close, nothing too fancy. One thing that does is make sure that no more code is ran within the application, so the `for` loop for example isn't ran at all if `Process.arguments.count == 1`.
+
+`Process` comes from the Swift language and represents the process that your app is running in, think of it as a swimming lane that our app is swimming in. We want to look at how many things were passed in to our app from the command line, and `Process` gives us this information.
+
+### Loops
+
+Let's try unpack what we're seeing here. Jory is expecting that he would get an array like this:
+
+```
+var arguments = ["stampicon", "inputfile.png", "--output", "outputfile.png", "--text", "text to overlay"]
+```
+
+So this app want's to look through the arguments and then compare what they find, then use that to generate a configuration object. We'll use the above array as an example.
+
+``` swift
+for idx in 1..<Process.arguments.count {
+
+    let argument = Process.arguments[idx]
+
+    if idx == 1 {
+        config.inputFile = argument
+        continue
+    }
+
+    switch argument {
+    case "--text":
+        config.text = Process.arguments[idx+1]
+
+    case "--output":
+        config.outputFile = Process.arguments[idx+1]
+
+    case "--font":
+        config.fontName = Process.arguments[idx+1]
+
+    case "--textcolor":
+        config.textColor = NSColor(rgba: Process.arguments[idx+1])
+
+    case "--badgecolor":
+        config.badgeColor = NSColor(rgba: Process.arguments[idx+1])
+
+    default:
+        continue
+    }
+}
+```
+
+OK, so, this will loop through a list of numbers going from 1 to the number of arguments minus 1, which in our case is 6 and assign them to `idx` - short for Index. So 1, 2, 3, 4, 5.
+
+Then we take that index, and pull out the string and call that `argument` with `let argument = Process.arguments[idx]`. Jory is using `let` here to inform others that this will never change.
+
+Then the index is checked to see if it is the first one, and if so take the string and call set that as the `config`'s `inputFile`.
+
+``` swift
+if idx == 1 {
+    config.inputFile = argument
+    continue
+}
+```
+
+It then uses `continue` to say, "now go on to the next thing in the `for` loop." So it will start again with `idx` being `2`.
+
+So with `idx` as 2, it would go past that `if` and get to a `switch`. We've not looked at switches, so lets make a playground, and have a look.
+
+You can make one in the current project by going to the menubar, Hitting `File` > `New` > `Playground`. Then making sure you choose OS X.
+
+Then add this.
+
+``` swift
+var number = 1
+switch number {
+case 1:
+    print("hi")
+case 2:
+    print("bye")
+default:
+    print("what?")
+}
+```
+
+If you remember how a `Bool` can represent on and off, and then an `enum` can be the same kind of idea but with more states. Well `if` represents something being on or off and `switch` represents multiple states.
```

Comment:
First sentence sounds awkward. I like the analogue between if : switch :: Bool : enum a lot. I think it just needs to be expanded out a bit.